sqlalchemy_ibmi/base.py

```diff
@@ -749,50 +749,10 @@ def create_connect_args(self, url):
         opts = url.translate_connect_args(username='user')
         opts.update(url.query)
         keys = opts
-
-        connect_args = {}
-        for param in ('ansi', 'unicode_results', 'autocommit'):
-            if param in keys:
-                connect_args[param] = util.asbool(keys.pop(param))
-
-        if 'odbc_connect' in keys:
-            connectors = [urllib.parse.unquote_plus(keys.pop('odbc_connect'))]
-        else:
-            dsn_connection = 'dsn' in keys or \
-                             ('host' in keys and 'database' not in keys)
-            if dsn_connection:
-                connectors = ['dsn=%s' % (keys.pop('host', '') or
-                                          keys.pop('dsn', ''))]
-            else:
-                connectors = [
-                    "DRIVER={%s}" %
-                    keys.pop('driver', self.pyodbc_driver_name),
-                    'System=%s' % keys.pop('host', ''),
-                    'DBQ=QGPL',
-                    "PKG=QGPL/DEFAULT(IBM),2,0,1,0,512"
-                ]
-                db_name = keys.pop('database', '')
-                if db_name:
-                    connectors.append("DATABASE=%s" % db_name)
-
-            user = keys.pop("user", None)
-            if user:
-                connectors.append("UID=%s" % user)
-                connectors.append("PWD=%s" % keys.pop('password', ''))
-            else:
-                connectors.append("trusted_connection=yes")
-
-            # if set to 'Yes', the ODBC layer will try to automagically convert
-            # textual data from your database encoding to your client encoding
-            # This should obviously be set to 'No' if you query a cp1253
-            # encoded database from a latin1 client...
-            if 'odbc_autotranslate' in keys:
-                connectors.append(
-                    "AutoTranslate=%s" %
-                    keys.pop("odbc_autotranslate"))
-
-            connectors.extend(['%s=%s' % (k, v) for k, v in keys.items()])
-        return [[";".join(connectors)], connect_args]
+        args = ['host', 'user', 'password', 'ansi', 'attrs_before',
+                'autocommit', 'encoding', 'readonly', 'timeout']
+        keys = {key: keys[key] for key in keys if key in args}
+        return [["Driver={%s}" % self.pyodbc_driver_name], keys]
```

Comment:
This looks good, but I don't think we should I don't think we should support encoding, ansi, or attrs_before.

Reply:
I'm thinking now maybe we want to raise an error instead of just removing unknown keys. In that case, perhaps something like

```python
allowed_options = {'host', 'user', ... }
if not allowed_options >= keys.keys():
    # raise error here?
...
```

I'm not sure whether other dialects do such things, though.